### ISSUE TITLE
add support to other build types than only debug and release 

### ISSUE THREAD

**Issue by kibotu** [airbnb/Showkase#274]
Currently only debug and release build types are supported

imagine having a setup with build types

```groovy

android {

    buildTypes {

        alpha {
            initWith(buildTypes.debug)
        }

        release {
        }

        beta {
            initWith(buildTypes.release)
        }
    }

    dependencies {
        alphaImplementation "com.airbnb.android:showkase:1.0.0-beta14"
        kaptAlpha "com.airbnb.android:showkase-processor:1.0.0-beta14"
    }
}

```

then the dependencies won't be available in alpha builds because the way it is published to maven central 

https://repo1.maven.org/maven2/com/airbnb/android/showkase/1.0.0-beta14/

compared to e.g. leakcanary 

https://repo1.maven.org/maven2/com/squareup/leakcanary/leakcanary-android/2.9.1/

the problem lies with the file naming 

where instead of having one artifact, the showkase library has 2: one for debug, which is not a bad idea if you want automatic switching with just adding implementation instead of having to add **debugImplementation** and **releaseImplementation**

however I'm not aware of a way to have those files added for alpha/beta/release or any other non default build type now

--- 

for this to work we would have to have two different artifacts, e.g. one for 

```groovy
    com.airbnb.android:showkase:1.0.0-beta14 
    com.airbnb.android:showkase-no-op:1.0.0-beta14 // e.g. for release 
```

**Comment by kibotu** [comment]
ok turns out there is a workaround

```groovy
android {

    buildTypes {

        alpha {
            matchingFallbacks = ['debug']
        }
    }
}
            
```

**Comment by popovanton0** [comment]
I also encountered this problem, spent a whole day on it, and even wrote a [Medium article](https://medium.com/@popovanton0/fixing-dependency-metadata-in-gradle-65fd86abe4e1) about solving it.

TLDR; As it turns out, if you have a project like this:
```mermaid
graph LR
  B(base-ui) --> D("design-system<br>(with showkase)")
  A(app) --> A1(feature-module) --> B
  A1 --> D
```
you need (not entirely sure why) to specify `matchingFallbacks` in all `build.gradle` files. I did this using a root file that applies to all modules the project.

Unfortunately, I haven’t realized that need in time and wrote _almost_ a [Gradle plugin](https://gist.github.com/popovanton0/f73631c6a819c630d543c10f8e42fff3#file-gradlemetadatarule-gradle-kts) to modify dependency metadata.

Why the library has multiple versions (`debug` and `release`)? Is it necessary? If not, it will be mush easier for developers if the library had only one version, like the aforementioned LeakCanary.

**Comment by vinaygaba** [comment]
Thanks, I do see the problem and I don't think it's intended to be like this. I think having a single artifact makes sense to me. I can fix this in the next release.

**Comment by vinaygaba** [comment]
@popovanton0 also, thank you for the write up and documenting a way to unblock people until the fix is merged

**Comment by vinaygaba** [comment]
Published a PR to fix this - https://github.com/airbnb/Showkase/pull/291
Not 100% sure I have the right fix so if either of you have thoughts, please chime in on the PR
